app/models/talentlms/facade.rb

```diff
@@ -10,42 +10,111 @@ def initialize
       @api = Config.first
     end
 
+    def talentlms_user_record_by_id(id)
+      result = @api.post('users', { id: id })
+      result
+    rescue RuntimeError => e
+      raise e unless e.message.include?('The requested user does not exist')
+    end
+
+    def talentlms_user_record_by_email(email)
+      result = @api.post('users', { email: email })
+      result
+    rescue RuntimeError => e
+      raise e unless e.message.include?('The requested user does not exist')
+    end
+
+    def talentlms_user_record_by_username(username)
+      result = @api.post('users', { username: username })
+      result
+    rescue RuntimeError => e
+      raise e unless e.message.include?('The requested user does not exist')
+    end
```

Comment:
I think the idea with naming methods in the facade was that they would be `lms_*` rather than specific to the LMS of the facade, that should mean you could rely on the names of the methods (or the API) to not change between facades.  Acknowledging it means rewriting code further down, can we do:
   ```suggestion
    def lms_user_record_by_id(id)
      result = @api.post('users', { id: id })
      result
    rescue RuntimeError => e
      raise e unless e.message.include?('The requested user does not exist')
    end

    def lms_user_record_by_email(email)
      result = @api.post('users', { email: email })
      result
    rescue RuntimeError => e
      raise e unless e.message.include?('The requested user does not exist')
    end

    def lms_user_record_by_username(username)
      result = @api.post('users', { username: username })
      result
    rescue RuntimeError => e
      raise e unless e.message.include?('The requested user does not exist')
    end
```

Reply:
alternatively, maybe these should be private and then I don't care.

Reply:
Updated with this commit https://github.com/greenriver/hmis-warehouse/pull/4830/commits/b4cced473718c2a9835eb18fe8cb27462b8f5821